source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,2397 @@
+which is always nested under ``defaultTransactionOptions``. This is a special
+case where all drivers represent TransactionOptions as a separate struct/object,
+even if they do not do so for ``startTransaction``.
+
+
+Change Stream Tests
+-------------------
+
+
+Error Assertions
+~~~~~~~~~~~~~~~~
+
+`Iterating the Change Stream <../change-streams/tests#iterating-the-change-stream>`__
+in the change stream spec test documentation states:
+
+  If the test expects an error and one was not thrown by either creating the
+  change stream or executing the test's operations, iterating the change stream
+  once allows for an error to be thrown by a getMore command.
+
+It's not clear if this is necessary because the existing change stream spec test
+format does not differentiate between errors raised by creating or iterating the
+change stream. For instance, an invalid pipeline operator (error code 40324)
+should be raised by the initial ``aggregate`` command.
+
+Since the unified format would allow tests to differentiate between the initial
+``watch`` operation and `iterateUntilDocumentOrError`_ operation, and each
+can expect its own error, this may not be a concern; however, if the purpose of
+the quoted text above to to account for drivers that may not even execute the
+``aggregate`` command at the time ``watch`` is called (instead deferring it to
+the first iteration of ChangeStream), we may still need to account for that in
+the unified test format.
+
+If so, one possible solution may be to document that ``watch`` operations
+expecting an error must also iterate the returned change stream once if the
+operation itself did not initially raise an error. Of course, we should confirm
+that this (or any) solution is sufficient for all drivers.
+
+
+Command Monitoring Assertions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The change stream spec is also unique among other specs in that its event
+assertions only consider as many events as were expected and ignore both extra,
+observed events (e.g. getMore) and an optional killCursors (which SHOULD be
+issued when resuming). The issue of extra getMore events may pertain to either
+sharding or have something to do with variation among drivers in how ``watch``
+is implemented (e.g. eagerly calling ``aggregate`` vs. deferring until the first
+iteration).
+
+A possible solution to address this need may be to extend
+`test.expectedEvents <test_expectedEvents_>`_ to allow a list of command names
+to be specified, for which any command monitoring events will be ignored (as we
+currently do for security events and ``configureFailPoint``). If that addresses
+*both* needs, we may not need anything more; however, there are some change
+stream tests that *do* expect getMore, so in such a case we may still need a
+separate `test.expectedEvents <test_expectedEvents_>`_ option to instruct the
+test runner to only assert as many observed events are expected and ignore any
+additional events.
+
+
+getResumeToken
```

Comment:
I think we can hold off on this until it is needed?

Reply:
SGTM. Will leave this open until we meet Wednesday to go through team review.